From 17238a89cbaa0f17f3c4f4e8540457f33208431d Mon Sep 17 00:00:00 2001
From: Uladzislau Rezki <urezki@gmail.com>
Date: Sun, 16 Oct 2022 16:23:02 +0000
Subject: [PATCH 08/20] FROMGIT: scsi/scsi_error: Use call_rcu_hurry() instead
 of call_rcu()

Slow boot time is seen on KVM running typical Linux distributions due to
SCSI layer calling call_rcu(). Recent changes to save power may be
causing this slowness. Using call_rcu_hurry() fixes the issue and brings
the boot time back to what it originally was. Convert it.

(cherry picked from commit 716cac6652c8512ba939a7f62eada6589d9c84d6
 https://git.kernel.org/pub/scm/linux/kernel/git/paulmck/linux-rcu.git rcu/next)

Bug: 258241771
Bug: 222463781
Test: powerIdle lab tests.
Tested-by: Joel Fernandes (Google) <joel@joelfernandes.org>
Signed-off-by: Uladzislau Rezki <urezki@gmail.com>
Signed-off-by: Joel Fernandes (Google) <joel@joelfernandes.org>
Change-Id: I968164f96008e23e5d249ccd99224a1feca2a38c
Signed-off-by: Joel Fernandes <joelaf@google.com>
Reviewed-on: https://chromium-review.googlesource.com/c/chromiumos/third_party/kernel/+/3965076
Reviewed-by: Ross Zwisler <zwisler@google.com>
(cherry picked from commit b54c7f53a988395e29885c74c69531ba3148e835)
[Cherry picked from chromeos-5.10 tree. Minor tweaks to commit message
to match Android style]
Signed-off-by: Qais Yousef <qyousef@google.com>

Upstream: 54d87b0a0c19bc3f740e4cd4b87ba14ce2e4ea73 ("scsi/scsi_error: Use call_rcu_hurry() instead of call_rcu()")
ACK android13-5.10: 48fcc3fe2017b4592cf547e069bdd80bc15c29c8, taken from arter97/master
Backport notes: call_rcu_flush() renamed to call_rcu_hurry().
---
 drivers/scsi/scsi_error.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/scsi_error.c b/drivers/scsi/scsi_error.c
index 3d3d139127ee..c02fadcaffce 100644
--- a/drivers/scsi/scsi_error.c
+++ b/drivers/scsi/scsi_error.c
@@ -276,7 +276,7 @@ void scsi_eh_scmd_add(struct scsi_cmnd *scmd)
 	 * Ensure that all tasks observe the host state change before the
 	 * host_failed change.
 	 */
-	call_rcu(&scmd->rcu, scsi_eh_inc_host_failed);
+	call_rcu_hurry(&scmd->rcu, scsi_eh_inc_host_failed);
 }
 
 /**
